{{cookiecutter.project_slug}}/README.md

```diff
@@ -37,6 +37,9 @@
 {% if cookiecutter.readthedocs == "Yes" -%}
 [![Documentation Status](https://readthedocs.org/projects/{{ remote_slug }}/badge/)](https://{{ remote_slug }}.readthedocs.io/)
 {% endif -%}
+{% if cookiecutter.codecovio == "Yes" -%}
+[![codecov](https://codecov.io/{% if is_github %}gh{% endif %}{% if is_gitlab %}gl{% endif %}/{{ username }}/{{ remote_slug }}/branch/main/graph/badge.svg)](https://codecov.io/{% if is_github %}gh{% endif %}{% if is_gitlab %}gl{% endif %}/{{ username }}/{{ remote_slug }})
```

Comment:
just for readability,would it be possible to replace this `{% if is_github %}gh{% endif %}{% if is_gitlab %}gl{% endif %}` construction with something like `{{ git_provider }}`, i.e. some variable that is previously set to either `gh` or `gl` depending on the git provider? or is that not possible?

Reply:
It is possible, unfortunately the scope of such variables does not go beyond the current file though. Will do nevertheless. FWIW, there is an upstream PR at cookiecutter that will make these things very nice, because it allows cookiecutters to ship a local Jinja2 extension, which will make it possible to shift all that logic into Python code and keep the template language code very simple.